z3/src/tactic.rs

```diff
@@ -0,0 +1,159 @@
+use std::ffi::{CStr, CString};
+use std::fmt;
+
+use z3_sys::*;
+use Context;
+use Goal;
+use Params;
+use Tactic;
+use Z3_MUTEX;
+
+impl<'ctx> Tactic<'ctx> {
+    pub fn list_all(ctx: &'ctx Context) {
+        let p = unsafe {
+            Z3_get_num_tactics(ctx.z3_ctx)
+        };
+        for n in 0..p {
+            let t = unsafe {
+                Z3_get_tactic_name(ctx.z3_ctx, n)
+            };
+            match unsafe { CStr::from_ptr(t) }.to_str() {
+                Ok(s) => println!("{}", s),
```

Comment:
did you mean to print this out? maybe we should be returning `Vec<String>` or `impl Iterator<Item=String>` here